datalad/support/handlerepo.py

```diff
@@ -288,4 +290,84 @@ def get_metadata(self, files=None):
         return joined_graph
 
     def set_metadata(self, graph, msg="Metadata saved."):
-        raise NotImplementedError
\ No newline at end of file
+        raise NotImplementedError
+
+    def import_metadata(self, importer, files=None, data=None,
+                        about_uri=DLNS.this):
+        """Imports metadata to the datalad-conform metadata representation
+        of the handle.
+
+        Expects either `files` or `data` to be not `None`.
+
+        Parameters:
+        ___________
```

Comment:
should be dashes, not underscores AFAIK.  it is 'the numpy docstring' format https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt